### PR TITLE
N+1 Query fix & Integration tests

### PR DESCRIPTION
**Pull request by Thoth1111**
In this branch, the following was achieved:
✅ Modify `post` index and show html.erb files to fix the `N+1` query problem.
✅ Add integration tests for `user`. Done by collaborator.
✅ Add integration tests for `post` and refactor code to ensure the test objectives are met. Delete obsolete `request` tests for placeholder text.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
-    @posts = Kaminari.paginate_array(Post.all).page(params[:page]).per(5)
     @user = User.find(params[:user_id])
+    @posts = @user.posts.page(params[:page]).per(5)
```
Comment (julie-ify) [inline app/controllers/posts_controller.rb]: - [x] The comments for the posts should be fetched with them. Kindly modify this method further to fetch comments, Here is a basic example of the query you can work with:

  ```
  @posts = @user.posts.includes(:comments)
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Thoth1111 and @stevenmukama,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Thoth1111) [reply app/controllers/posts_controller.rb]: Thank you. I've modified the `index` method of `posts_controller` to fetch comments with the users posts in my latest commit.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
-    @posts = Kaminari.paginate_array(Post.all).page(params[:page]).per(5)
     @user = User.find(params[:user_id])
+    @posts = @posts = @user.posts.includes(:comments).page(params[:page]).per(5)
```
Comment (julie-ify) [inline app/controllers/posts_controller.rb]: - [x] Kindly remove one `@posts = `. You should have something like this:

  ```
  @posts = @user.posts.includes(:comments).page(params[:page]).per(5)
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @Thoth1111 and @stevenmukama,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Thoth1111) [reply app/controllers/posts_controller.rb]: My mistake. I've fixed it now.

**Review by cynthiainga — APPROVED** [review]
Hi @Thoth1111, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] Correct Gitflow ✔️ 
- [ ] No linters errors ✔️ 
- [ ] Tests are passing ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Thoth1111** [comment]
Thank you.

**Pull request merged**